src/Core/MySQL/MySQLUtils.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Common/assert_cast.h"
+#include "Columns/ColumnNullable.h"
+#include "Columns/ColumnsDateTime.h"
+#include "Core/DecimalFunctions.h"
+#include "DataTypes/IDataType.h"
+#include "base/types.h"
+
+namespace DB
+{
+namespace MySQLProtocol
+{
+namespace MySQLUtils
+{
+
+DecimalUtils::DecimalComponents<DateTime64>
+getNormalizedDateTime64Components(const DataTypePtr data_type, const ColumnPtr col, const int row_num)
```

Comment:
Let's pass a `size_t` (just in case a part grows > 2 billion rows which sounds not too far fetched)

Reply:
I'll do that in one of my PRs

Reply:
I agree with this, I should've fixed it earlier, as that's how it was originally in ProtocolText, so that is why I kept using it in new implementations. Let's fix both ResultSetRow for text/binary and utils, then :+1: